src/lib/elementary/meson.build

```diff
@@ -958,10 +957,16 @@ elementary_src = files([
   'efl_ui_spotlight_animation_manager.c',
 ])
 
-elementary_deps = [emile, eo, efl, edje, ethumb, ethumb_client, emotion, ecore_imf, ecore_con, eldbus, efreet, eio, buildsystem]
+elementary_deps = [emile, eo, efl, edje, emotion, ecore_imf, ecore_con, eldbus, efreet, eio, buildsystem]
 elementary_pub_deps = [eina, eet, evas, ecore, ecore_evas, ecore_file, ecore_input, ecore_imf, ecore_con,
-                       edje, eldbus, efreet, ethumb_client, efl]
+                       edje, eldbus, efreet, efl]
 elementary_ext_deps = [atspi, dl, intl, buildsystem_simple]
+
+if get_option('ethumb')
+  elementary_src += files('elm_thumb.c')
+  elementary_deps += [ethumb, ethumb_client]
+  elementary_pub_deps += ethumb_client
+endif
```

Comment:
I was thinking here, what is more..."mesonic" or would sound better?

1. Initialize elementary_deps without the `ethumb{,_client}` and add it inside the `if get_option('ethumb')` (i.e. how you proposed); or
2. Initialize elementary_deps with `ethumb{,_client}` inside (i.e. how it was before the change), which can be the dependency itself or an empty `declare_dependency`.

Reply:
hmmm, I prefer the first one, but I'm not much of a meson guy...
If we choose the second we need to change a little `meson.build` to generate the empty declarations. It wouldn't be a big change either, but it could lead us to some hard debugging in the future.

Reply:
Hum, fair.

Well, we already have to filter source-files, soooooooooo ok!